tests/evaluate/test_default_evaluator.py

```diff
@@ -2990,6 +2983,174 @@ def test_default_metrics_as_custom_metrics_static_dataset():
     assert "exact_match/v1" in results.metrics.keys()
 
 
+def test_derived_metrics():
+    def metric_1(predictions, targets, metrics):
+        return MetricValue(
+            scores=[0, 1],
+            justifications=["first justification", "second justification"],
+            aggregate_results={"aggregate": 0.5},
+        )
+
+    def metric_2(predictions, targets, metrics, metric_1):
+        return MetricValue(
+            scores=[score * 5 for score in metric_1.scores],
+            justifications=[
+                "metric_2: " + justification for justification in metric_1.justifications
+            ],
```

Comment:
this looks okay to me, but curious how we expect users to combine justifications? scores make sense to me. what is our recommended CUJ for this.

Reply:
Probably users will not be using the justifications of other metrics to come up with justifications for a derived metric - I'd imagine that they would come up with some entirely new justification or not have a justification at all. Here I wanted to test that they could access other metrics justifications in the case that they wanted to, not sure how often users would actually want to do that.